kotest-framework/kotest-framework-multiplatform-plugin-gradle/src/main/kotlin/io/kotest/framework/multiplatform/gradle/KotestMultiplatformCompilerGradlePlugin.kt

```diff
@@ -1,105 +1,87 @@
     */
-   private val version: String? by lazy {
-      val versionFromExtension = extension?.compilerPluginVersion?.orNull
-      if (versionFromExtension != null) {
-         println("Kotest compiler plugin [$versionFromExtension]")
-         return@lazy versionFromExtension
-      }
-
-      val engineDep = engineDeps().firstOrNull() ?: error("Cannot determine Kotest compiler plugin version if no Kotest engine dependencies are present")
-      val version = engineDep.version ?: return@lazy null
+   private var kotestExtension: KotestPluginExtension? = null
 
-      if (version.contains("LOCAL")) {
-         println("Detected dev engine version [$version]")
-      }
-
-      return@lazy version
+   override fun apply(target: Project) {
+      kotestExtension = target.createKotestExtension()
    }
 
-   private fun engineDeps(): List<Dependency> {
-      val project = target ?: error(missingProjectValError)
-
-      return project.configurations
-         .flatMap { it.all }
-         .flatMap { it.dependencies }
-         .filter { it.group == KotestGroupId && it.name.startsWith(engineDepPrefix) }
+   private fun Project.createKotestExtension(): KotestPluginExtension {
+      return extensions.create<KotestPluginExtension>(kotestPluginExtensionName).apply {
+         kotestCompilerPluginVersion.convention(KOTEST_COMPILER_PLUGIN_VERSION)
+      }
    }
 
    override fun getCompilerPluginId() = compilerPluginId
 
    override fun getPluginArtifact(): SubpluginArtifact =
-      SubpluginArtifact(KotestGroupId, KotestEmbeddableCompilerArtifactId, version)
+      SubpluginArtifact(
+         KotestGroupId,
+         KotestEmbeddableCompilerArtifactId,
+         kotestExtension?.kotestCompilerPluginVersion?.orNull,
+      )
 
    // This will soon be deprecated and removed, see https://youtrack.jetbrains.com/issue/KT-51301.
    override fun getPluginArtifactForNative(): SubpluginArtifact =
-      SubpluginArtifact(KotestGroupId, KotestNativeArtifactId, version)
+      SubpluginArtifact(
+         KotestGroupId,
+         KotestNativeArtifactId,
+         kotestExtension?.kotestCompilerPluginVersion?.orNull,
+      )
 
    override fun isApplicable(kotlinCompilation: KotlinCompilation<*>): Boolean {
+      val project = kotlinCompilation.target.project
+      val kotestExtension = project.extensions.findByType<KotestPluginExtension>()
+         ?: error("Could not find Kotest extension in $project")
+
       return when {
-         // if we can't find a version to use then we won't apply to this module
-         engineDeps().isEmpty() -> {
-            println("Warning: Kotest plugin has been added to project $target, but the project does not contain a Kotest engine dependency. Kotest will not be enabled.")
+         !kotestExtension.kotestCompilerPluginVersion.isPresent -> {
+            logger.warn("Warning: Kotest plugin has been added to $project, but could not determine Kotest engine version. Kotest will not be enabled.")
```

Comment:
For this case, should we explain that the version on the extension has been set to `null`, and that a possible way to fix the problem is to set an explicit version on the `kotest` extension?

Reply:
This error should only happen if the property is explicitly set to something nullable in a `build.gradle.kts`, and the stacktrace would lead here, so I think the wording isn't very important. So if you want to suggest something then I'll update it :)

Reply:
How about `Warning: the Kotest plugin has been added to $project, but kotestCompilerPluginVersion on the kotest extension has been set to null. Kotest will not be enabled for this project.`?